File: BankatiProject_Backend/services/User/src/main/java/com/example/user/users/response/UserResponse.java
package com.example.user.users.response;

import com.example.user.users.entity.Role;

import java.util.Date;

public record UserResponse(
        String id,
        String agentId,
        String firstName,
        String lastName,
        String email,
        String address,
        String cin,
        Date birthDate ,
        String phoneNumber,

        Role role,
        String password,
        boolean isFirstLogin ,
        String commercialRn,
        String image,
        String patentNumber ,
        Boolean isPaymentAccountActivated ,
        String typeHissab,
        String currency

) {
}

File: BankatiProject_Front-end/src/app/models/UserResponse.ts
export interface UserResponse {
  id: string;
  agentId: string | null;
  firstName: string;
  lastName: string;
  email: string;
  address: string | null;
  cin: string | null;
  birthDate: string | null; // ou un type Date si vous préférez
  phoneNumber: string;
  role: string; // Peut être 'ADMIN', 'AGENT', ou 'CLIENT'
  password: string; // (si vous en avez besoin, mais pensez à éviter de l'exposer dans les réponses)
  isFirstLogin: boolean;
  commercialRn: string | null;
  image: string | null;
  patentNumber: string | null;
  isPaymentAccountActivated: boolean | null;
  typeHissab: string | null;
  currency: string | null;
}
